src/screens/Exchange/ExchangeConfirm.js

```diff
@@ -115,6 +118,16 @@ class ExchangeConfirmScreen extends React.Component<Props, State> {
     transactionSpeed: NORMAL,
   };
 
+  componentDidUpdate() {
+    const {
+      executingExchangeTransaction,
+      navigation,
+    } = this.props;
+    if (!executingExchangeTransaction) {
+      navigation.goBack();
```

Comment:
How could we end up on this screen if we're not supposed to be here? Can we handle this somewhere else?

Reply:
This is because we're navigation to asset send flow and we want to dismiss this screen once back from that flow and transaction succeeded (we dispatch exchange transaction reset on succeed).